openstf/model/objective.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+import optuna
+
+from openstf.enums import MLModelType
+from openstf.model.model_creator import ModelCreator
+from openstf.model_selection.model_selection import split_data_train_validation_test
+from openstf.metrics.metrics import mae
+
+EARLY_STOPPING_ROUNDS: int = 10
+TEST_FRACTION: float = 0.1
+VALIDATION_FRACTION: float = 0.1
+# See https://xgboost.readthedocs.io/en/latest/parameter.html for all possibilities
+EVAL_METRIC: str = "mae"
+
+# https://optuna.readthedocs.io/en/stable/faq.html#objective-func-additional-args
+class XGBRegressorObjective:
+    """XGBRegressor optuna objective function.
+
+        Use this class for optimization using an optuna study. The constructor is used to
+        set the "input_data" and optionally add some configuration. Next the instance
+        will be called by he optuna study during optimization.
+
+        Example:
+            # initialize the objective function
+            objective = XGBRegressorObjective(input_data, test_fraction)
+            # use the objective function
+            study.optimize(objective)
+    """
+
+    MODEL_TYPE = MLModelType.XGB
+
+    def __init__(
+        self,
```

Comment:
Here you could add type hints as well

Reply:
I'm not totally sure how useful that is in this case. It really obvious that a constructor always return an instance of the class. Moreover, I'm not even sure its possible? My editor complaints since the class is not yet defined:
![image](https://user-images.githubusercontent.com/6715707/122378889-a3ffdb80-cf66-11eb-81b9-9d227797bcc3.png)